File: src/main/java/com/sudosoo/takeiteasy/common/NumericValidator.java
package com.sudosoo.takeiteasy.common;

import com.sudosoo.takeiteasy.common.annotation.CustomDateFormatValidator;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class NumericValidator implements ConstraintValidator<CustomDateFormatValidator, String> {

    @Override
    public void initialize(CustomDateFormatValidator constraintAnnotation) {
    }

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return value != null && value.matches("\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}") && value.length() == 16;
    }
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/application/annotation/NumericValidator.kt
package com.sudosoo.takeItEasy.application.annotation

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class NumericValidator : ConstraintValidator<CustomDateFormatValidator?, String?> {
    override fun initialize(constraintAnnotation: CustomDateFormatValidator?) {
    }

    override fun isValid(value: String?, context: ConstraintValidatorContext): Boolean {
        return value != null && value.matches("\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}".toRegex()) && value.length == 16
    }
}
